lib/cart/widgets/cart_info_card.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flex_storefront/flex_ui/tokens/colors.dart';
+import 'package:flex_storefront/flex_ui/tokens/sizes.dart';
+import 'package:flutter/material.dart';
+
+enum CartInfoType {
+  info,
+  success,
+  warning,
+  error,
+}
+
+class CartInfoCard extends StatelessWidget {
+  const CartInfoCard({
+    super.key,
+    this.type = CartInfoType.info,
+    required this.message,
+  });
+
+  final CartInfoType type;
+  final String message;
+
+  Color getCardColor() {
+    switch (type) {
+      case CartInfoType.success:
+        return FlexColors.success;
+      case CartInfoType.warning:
+        return FlexColors.warning;
+      case CartInfoType.error:
+        return FlexColors.error;
+      case CartInfoType.info:
+      default:
+        return FlexColors.info;
+    }
+  }
+
+  Color getTextColor() {
+    switch (type) {
+      case CartInfoType.success:
+        return FlexColors.onSuccess;
+      case CartInfoType.warning:
+        return FlexColors.onWarning;
+      case CartInfoType.error:
+        return FlexColors.onError;
+      case CartInfoType.info:
+      default:
+        return FlexColors.onInfo;
+    }
+  }
```

Comment:
There should be no default clauses on those enums. So when we add a new value to the enum, we have compile errors that point where an update is necessary.

Also, those characteristics (`textColor` and `cardColor`) could be attributes of the enum 🙂 I find it more elegant that way (and more scalable):

```dart
enum CartInfoType {
  info(
    cardColor: FlexColors.info,
    textColor: FlexColors.onInfo,
  ),
  success(
    cardColor: FlexColors.success,
    textColor: FlexColors.onSuccess,
  ),
  warning(
    cardColor: FlexColors.warning,
    textColor: FlexColors.onWarning,
  ),
  error(
    cardColor: FlexColors.error,
    textColor: FlexColors.onError,
  );

  const CartInfoType({
    required this.cardColor,
    required this.textColor,
  });

  final Color cardColor;
  final Color textColor;
}
```